lib/TabTracker.js

```diff
@@ -45,6 +52,16 @@ TabTracker.prototype = {
     }
     tabs.removeListener("open", this.onOpen);
     Services.obs.removeObserver(this.onPerfLoadComplete, PERF_LOG_COMPLETE_NOTIF);
+    Services.obs.removeObserver(this.onCacheUpdate, CACHE_UPDATE_NOTIF);
```

Comment:
A good side effect of this is that it makes for clearer and more explicit code.

Should we start needing database queries in other modules than `ActivityStreams.js`, it makes sense to create a `PlacesQueryService` or similarly named object which will be exported as a singleton instance whose purpose is to keep track of caching Places queries. It could even be exported from `PlacesProvider`.

Thoughts?